### DISCUSSION TITLE
Compilation for CentOS x64

### DISCUSSION THREAD

**Discussion by sayyedaleem** [scsitape/stenc#101] [General]
Hi community,

I think Stenc compilation package does not run CentOS Linux release 7.6.1810 (Core). Running autogen.sh command results in the following error;
./autogen.sh: line 6: autoreconf: command not found

Do I need to install additional package before compiling Stenc on CentOS Linux release 7.6.1810 (Core).

Thanks
Aleem

**Comment by jonasstein** [comment]
it should be part of the autoconf package. Is it installed?
Can you find autoreconf with
`which autoreconf`

see also https://www.thegeekdiary.com/autoreconf-command-not-found/

**Reply by sayyedaleem** [reply]
It seems that configure.ac has a syntax error when executing 

[root@labvtl stenc]# **./autogen.sh && ./configure**
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking dependency style of gcc... gcc3
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for sys/types.h... (cached) yes
checking sys/machine.h usability... no
checking sys/machine.h presence... no
checking for sys/machine.h... no
checking for g++... no
checking for c++... no
checking for gpp... no
checking for aCC... no
checking for CC... no
checking for cxx... no
checking for cc++... no
checking for cl.exe... no
checking for FCC... no
checking for KCC... no
checking for RCC... no
checking for xlC_r... no
checking for xlC... no
checking whether we are using the GNU C++ compiler... no
checking whether g++ accepts -g... no
checking dependency style of g++... none
checking for egrep... (cached) /usr/bin/grep -E
checking whether to output raw SCSI messages... no
checking default encryption algorithm index to use... 0
checking your OS... Linux
checking whether to build with static libgcc... no
checking for pandoc... yes
_./configure: line 4643: syntax error near unexpected token `BASH_COMPLETION_DIR,'
./configure: line 4643: `PKG_CHECK_VAR(BASH_COMPLETION_DIR, bash-completion >= 2.0, completionsdir,'_

**Reply by sunwire** [reply]
> It seems that configure.ac has a syntax error when executing

This is because of the old version of pkgonfig

**Comment by sunwire** [comment]
CentOS 7 is too old, gcc-c++ doesn't support c++17, also pkgconfig is too old.

If you really need stenc, the last supported version is 1.0.8
To compile you have to install these packages autoconf automake gcc gcc-c++
